src/context/reducer.ts

```diff
@@ -0,0 +1,17 @@
+import { GlobalState, Types } from "./context.interface";
+
+export const rootReducer = (
+  state: GlobalState,
+  action: { type: Types; payload?: any }
```

Comment:
[Optional] We could define a type for the `payload`

Reply:
different action expect different payload so can't really define any particular type here.